idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/action/AnActionSyntax.kt

```diff
@@ -10,55 +10,113 @@ import com.intellij.openapi.actionSystem.TimerListener
 import com.intellij.openapi.application.ModalityState
 import javax.swing.Icon
 
-// TODO: Check Default of actionId
+/**
+ * [AnActionExtensionProvider] is in charge of the lifecycle of [AnAction].
+ * [AnAction] is an overly broad concept to access and execute nearly anything in the editor.
+ * Put differently, [AnAction] contains a computation which can be evaluated on-demand by the user.
+ * When a user triggers [AnAction] the [AnAction.actionPerformed] function is executed.
+ * [AnAction] can form starkly different purposes, from changing CompilerConfigurations in the editor to opening UI element's with media content.
+ * Here are a few Links to the Action System from IntelliJ's DevGuid:
+ * - [Plugin Actions](http://www.jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_actions.html?search=act)
+ * - [Executing and updating actions](http://www.jetbrains.org/intellij/sdk/docs/basics/action_system.html?search=act#executing-and-updating-actions)
+ */
 interface AnActionSyntax : AnActionUtilitySyntax {
+
+  /**
+   * Registers the [action] with [actionId] as it's identifier.
+   * ```kotlin:ank:playground
+   * import com.intellij.openapi.wm.ToolWindowManager
+   * import arrow.meta.ide.resources.ArrowIcons
+   * import arrow.meta.invoke
+   * import arrow.meta.Plugin
+   * import arrow.meta.ide.IdeMetaPlugin
+   * import com.intellij.openapi.project.Project
+   * import com.intellij.openapi.actionSystem.AnActionEvent
+   *
+   * val IdeMetaPlugin.exampleAction: Plugin
+   * get() = "Example Action" {
+   *   meta(
+   *   //sampleStart
+   *     addAnAction(
+   *      actionId = "Unique",
+   *      action = anAction(
+   *       icon = ArrowIcons.ICON1,
+   *       actionPerformed = { e: AnActionEvent ->
+   *        e.project?.let { project: Project -> ToolWindowManager.getInstance(project)?.getToolWindow("MetaToolWindow")?.activate(null) }
+   *        }
+   *       )
+   *      )
+   *   //sampleEnd
+   *    )
+   *  }
+   * ```
+   *
+   * This Action `Unique` has an Icon and opens a ToolWindow with a registered Id `MetaToolWindow`, assuming this ToolWindow is registered.
+   * User's are now able to search this Action through his [actionId]
```

Comment:
nit: `User's` -> `Users`?
The action in the sample code doesn't have a title, only an icon. Perhaps provide a value for `title` for completeness and easier searching in the actions, e.g. via `Help->Find Action`?

Reply:
Good call! Makes more sense. I couldn't find it without a title. I thought users can index Actions with `actionId`. Is that somehow possible @jansorg ?

Reply:
It‘s not possible, afaik. The id is used internally and not user visible, to my knowledge.